src/pybind/mgr/orchestrator/module.py

```diff
@@ -2367,3 +2384,10 @@ def _upgrade_stop(self) -> HandleCommandResult:
         completion = self.upgrade_stop()
         raise_if_exception(completion)
         return HandleCommandResult(stdout=completion.result_str())
+
+    @_cli_write_command('orch upgrade-image')
+    def _upgrade_service(self, service_type: NonCephImageServiceTypes, image: str) -> HandleCommandResult:
+        """Upgrade image for non-ceph image daemon"""
```

Comment:
This is a total nitpick, but I'd prefer the language used for this case to be `update` rather than `upgrade`. With our actual upgrade command we explicitly disallow going to an older version, whereas this command will take whatever image you want.